test/runtest.py

```diff
@@ -1511,6 +1514,47 @@ def wait_for_object_table():
 
     ray.worker.cleanup()
 
+  def testLogFileAPI(self):
+    ray.worker._init(redirect_output=True, num_local_schedulers=2,
+      start_ray_local=True)
+
+    @ray.remote
+    def say_hi():
+      print("hi")
+
+    say_hi.remote()
+    node_ip_address = ray.worker.global_worker.node_ip_address
+    log_files = ray.global_state.log_files()
+
+    self.assertEqual(isinstance(log_files[node_ip_address], dict), True)
+
+    # Make sure that "hi" appears in the log files
+    start_time = time.time()
+    found_hi = False
+    while time.time() - start_time < 10:
+      log_files = ray.global_state.log_files()
+      for ip, innerdict in log_files.items():
+        for filename, contents in innerdict.items():
+          contents_str = " ".join(str(x) for x in contents)
```

Comment:
I would do `"".join` instead of `" ".join`.

Also, `x` should ideally already be a string, so we could just do `"".join(contents)`